evm/gateway.js

```diff
@@ -55,6 +56,17 @@ const getSignedWeightedExecuteInput = async (data, operators, weights, threshold
     );
 };
 
+const fetchBatchData = async (apiUrl, batchId) => {
+    try {
+        const response = await httpGet(`${apiUrl}/${batchId}`);
```

Comment:
the helper isn't very useful rn since it's still making the caller construct the apiUrl. Instead, the inputs should be chain and batchId, and it should handle the call